### PR TITLE
fix:signurl err when uri include Chinese

### PR DESCRIPTION
**Pull request by diluga**
when i use s3cmd signurl with Chinese，i got this below，so make a patch to fix this！

```
s3cmd --version
s3cmd version 1.6.1

s3cmd signurl s3://s3demo/ceph知识树.png +200

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
    An unexpected error has occurred.
  Please try reproducing the error using
  the latest s3cmd code from the git master
  branch found at:
    https://github.com/s3tools/s3cmd
  and have a look at the known issues list:
    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions
  If the error persists, please report the
  following lines (removing any private
  info as necessary) to:
   s3tools-bugs@lists.sourceforge.net


You have encountered a UnicodeEncodeError.  Your environment
variable LANG=zh_CN.UTF-8 may not specify a Unicode encoding (e.g. UTF-8).
Please set LANG=en_US.UTF-8 or similar in your environment before
invoking s3cmd.

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

Invoked as: /usr/local/bin/s3cmd signurl s3://hxs1-s3soft/ceph知识树.png +200
Problem: UnicodeEncodeError: 'ascii' codec can't encode characters in position 31-33: ordinal not in range(128)
S3cmd:   1.6.1
python:   2.7.10 (default, Oct 23 2015, 19:19:21)
[GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.0.59.5)]
environment LANG=zh_CN.UTF-8

Traceback (most recent call last):
  File "/usr/local/bin/s3cmd", line 2923, in <module>
    rc = main()
  File "/usr/local/bin/s3cmd", line 2843, in main
    rc = cmd_func(args)
  File "/usr/local/bin/s3cmd", line 2017, in cmd_signurl
    debug("url to sign: %r" % url_to_sign)

UnicodeEncodeError: 'ascii' codec can't encode characters in position 31-33: ordinal not in range(128)

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
    An unexpected error has occurred.
  Please try reproducing the error using
  the latest s3cmd code from the git master
  branch found at:
    https://github.com/s3tools/s3cmd
  and have a look at the known issues list:
    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions
  If the error persists, please report the
  above lines (removing any private
  info as necessary) to:
   s3tools-bugs@lists.sourceforge.net
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

Signed-off-by: Feng Guo <diluga@gmail.com>

### CODE REVIEW

File: s3cmd [hunk 1]
```diff
@@ -2904,6 +2904,8 @@ if __name__ == '__main__':
         ## Our modules
         ## Keep them in try/except block to
         ## detect any syntax errors in there
+        reload(sys)
+        sys.setdefaultencoding('urf-8')
```
Comment (mdomsch) [inline s3cmd]: This can't be right - this isn't a valid encoding.

**Comment by mdomsch** [comment]
Thank you for attempting a fix.  The proposed fix cannot be correct, as it is setting an invalid encoding value 'urf-8' instead of 'utf-8', but utf-8 is already set in your environment LANG variable.  It's unclear how this patch fixes the problem.

**Comment by diluga** [comment]
i have set LANG=en_US.UTF-8,but it doesn't work, please  see the detail below:

```
─Diluga@DilugaMAC  ~
╰─$ env |grep LANG                                                                                                                                                               1 ↵
LANG=en_US.UTF-8
╭─Diluga@DilugaMAC  ~
╰─$ echo $LANG
en_US.UTF-8
╭─Diluga@DilugaMAC  ~
╰─$ s3cmd --version
s3cmd version 1.6.1
╭─Diluga@DilugaMAC  ~
╰─$ s3cmd signurl s3://hxs1-s3soft/EC性能对比.pdf +300

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
    An unexpected error has occurred.
  Please try reproducing the error using
  the latest s3cmd code from the git master
  branch found at:
    https://github.com/s3tools/s3cmd
  and have a look at the known issues list:
    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions
  If the error persists, please report the
  following lines (removing any private
  info as necessary) to:
   s3tools-bugs@lists.sourceforge.net


You have encountered a UnicodeEncodeError.  Your environment
variable LANG=en_US.UTF-8 may not specify a Unicode encoding (e.g. UTF-8).
Please set LANG=en_US.UTF-8 or similar in your environment before
invoking s3cmd.

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

Invoked as: /usr/local/bin/s3cmd signurl s3://hxs1-s3soft/EC性能对比.pdf +300
Problem: UnicodeEncodeError: 'ascii' codec can't encode characters in position 29-32: ordinal not in range(128)
S3cmd:   1.6.1
python:   2.7.10 (default, Oct 23 2015, 19:19:21)
[GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.0.59.5)]
environment LANG=en_US.UTF-8

Traceback (most recent call last):
  File "/usr/local/bin/s3cmd", line 2923, in <module>
    rc = main()
  File "/usr/local/bin/s3cmd", line 2843, in main
    rc = cmd_func(args)
  File "/usr/local/bin/s3cmd", line 2017, in cmd_signurl
    debug("url to sign: %r" % url_to_sign)
UnicodeEncodeError: 'ascii' codec can't encode characters in position 29-32: ordinal not in range(128)

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
    An unexpected error has occurred.
  Please try reproducing the error using
  the latest s3cmd code from the git master
  branch found at:
    https://github.com/s3tools/s3cmd
  and have a look at the known issues list:
    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions
  If the error persists, please report the
  above lines (removing any private

```

**Comment by diluga** [comment]
i am sorry to push the patch with "urf-8",i  have fix it,and push again!

File: s3cmd (hunk 1 above)
Comment (diluga) [inline s3cmd]: sorry，i fix it，push it again！

**Comment by fviard** [comment]
This issue should now be fixed in master with a more adequate fix:
https://github.com/s3tools/s3cmd/commit/8533cd191b426d13448a01f5c4d5b6f0dcb68d51
Anyway, thank you very much to have raised this issue that was never reported before.

**Comment by diluga** [comment]
thx!I have test this patch,it worked!

**Pull request closed without merging**